src/sysroot.rs

```diff
@@ -85,13 +85,25 @@ fn build_crate(
     }
 
     util::write(&td.join("Cargo.toml"), &stoml)?;
-    fs::copy(lockfile, &td.join("Cargo.lock")).with_context(|| {
+    let td_lockfile = &td.join("Cargo.lock");
+    fs::copy(lockfile, td_lockfile).with_context(|| {
         format!(
             "failed to copy Cargo.lock from `{}` to `{}`",
             lockfile.display(),
-            &td.join("Cargo.lock").display()
+            td_lockfile.display()
         )
     })?;
+    let mut perms = fs::metadata(&td_lockfile).map_err(|e| {
+        format!("failed to retrieve permissions for `{}`",
+            td_lockfile.display());
+        e
```

Comment:
I don't think that the `map_err` call here works as you're throwing away the `format!` result. Instead, I would recomment the [`with_context`](https://docs.rs/anyhow/1.0.34/anyhow/trait.Context.html#tymethod.with_context) method of `anyhow`:

```suggestion
    let mut perms = fs::metadata(&td_lockfile).with_context(|| {
        format!("failed to retrieve permissions for `{}`",
            td_lockfile.display())
```